src/map/entities/trustentity.cpp

```diff
@@ -84,7 +84,8 @@ void CTrustEntity::Die()
     CBattleEntity::Die();
     if (PMaster && PMaster->PPet == this && PMaster->objtype == TYPE_PC)
     {
-        petutils::DetachPet(PMaster);
+        CCharEntity* PChar = (CCharEntity*)PMaster;
+        PChar->RemoveTrust(this);
```

Comment:
once you begin testing trusts dying, it may be an issue having them go to Despawn state from inside the Die state.  Don't have to change it now I guess, but later we might want to make sure that they run their natural course of die -> despawn and just remove them from the vector (and party reload) without the other side effects